src/field/vecops.cxx

```diff
@@ -110,6 +126,10 @@ const Field2D Div(const Vector2D &v, CELL_LOC outloc) {
   Mesh *localmesh = v.x.getMesh();
   Field2D result(localmesh);
 
+  if (outloc == CELL_DEFAULT) {
+    outloc = v.getLocation();
```

Comment:
Note this could be a vector only location (VSHIFT) which should be invalid for getting/setting a field location.

Reply:
Good point -- is the "correct" output location well defined in that case? Maybe `CELL_CENTRE`?

Reply:
I'm not sure, @bendudson might have a better idea of what this should do in this case.

Reply:
Also, as above we might be able to just pass this through to the derivative routines to decide what to do. We can then take the location of the returned result to decide which coordinates location we want.

Reply:
I think it has to be `CELL_CENTRE` :+1: